### DISCUSSION TITLE
Fixedwing landing issues and hopeful improvements

### DISCUSSION THREAD

**Discussion by rts18** [iNavFlight/inav#10370] [Support] [answered]
@Scavanger @b14ckyy 
I have used the fixedwing automatic landing feature many times since its release. Finding it to work okay with aircraft that have a medium/high wing loading. But I've had little success with aircraft that have a medium/low wing loading; and not through lack of trying!
This type of aircraft tends to glide on farther and float more with a head wind, making it impossible to even land within 100m of the required location.
I am not referring to a glider, but common FPV planes commercially available.

I spent much time adjusting settings to force these aircraft down to the correct altitude when nearing the safehome landing marker. But to no avail. These settings make little to no difference when viewed in the logs, even when pushed to extremes. Like the ones with an asterisk.

`fwapproach 0 3800 50 0 -189 0 0`
`fwapproach 1 3800 50 1 -9 0 0`

`nav_fw_land_approach_length = 33000`
`nav_fw_land_final_approach_pitch2throttle_mod = 350` *
`nav_fw_land_glide_alt = 150` 
`nav_fw_land_flare_alt = 50`
`nav_fw_land_glide_pitch = -15` *
`nav_fw_land_flare_pitch = 3`
`nav_fw_min_thr = 1100` *

This is not a case of the estimated altitude being incorrect. I have watched many of these aircraft fly past the landing marker at the approximate altitude the OSD and telemetry is displaying e.g. 15m, even though the aircraft should be at 0m by that time.

___________________________________________
The second issue is related to having two safehome markers on opposite headings, within 80m of each other. With both being `Exclusive`. 
The reason for this; is to provide as much landing area as possible, within the boundaries of my landing site.

![landing area](https://github.com/user-attachments/assets/e2bee117-cd95-49a9-968d-edd714754e93)

The issue encountered with this layout is caused by having both safehomes that close together. The software gets confused when loitering to obtain a wind heading. It doesn't know which safehome to loiter around and gets stuck in a continuous loiter. 
___________________________________________
The third issue relates to using the landing safehome to loiter around when establishing a wind heading. This is far from ideal when flying at a club, when the rules state you're not permitted to fly over the pit area.
Its even a problem when your landing location has a tree, light post or power pole in the vicinity of the landing location. Because it limits how _low_ we can set the _approach altitude_, based on the height of those obstacles in the loiter location. Which in-turn effects the minimum we can set `nav_fw_land_approach_length`. And that comes full circle to the first issue I raised.

I know it may be said that my landing site is too small. But I don't believe this to be the case. I can manually land in this space with no issues. I have even had one aircraft perform a safe descending loiter landing in this area.

Any improvement would be greatly appreciated! thanks

**Comment by b14ckyy** [comment, marked as answer]
hi @rts18 and thanks for the detailed description of the issue. 
I have moved this to Discussions for now as its more a support topic. There are things that need to be indeed optimized for planes with high glide rationes and that was already a topic during development. But there are ways to counter this. 

First of all a thing for understanding: 2 safe homes that close together won't work. INAV always selects the one that is closest to the arming location, no matter what. As this one then is exclusive in heading, it will not be able to land if it has a tailwind for that landing location. Inav never swaps to the other safe home as only the one closest to the arming spot is loaded. This is decided when the craft is armed and will not change in flight. 

-----------------------

Now to your issue with the long glide and float. This is something that will be optimized in a future update to give a few more options to set. How exactly is not decided yet but we have some ideas. 
for now you can counteract that with a somewhat counter-intuitive method. 

If you look at the altitude pattern [graphic here](https://github.com/iNavFlight/inav/blob/master/docs/assets/images/Approach-Drawing-Side.png), you will see that the land Altitude dictates where the approach course intersects with the safe home location. But this only accounts for the powered flight phase. If the plane is very aerodynamic, even with motors cut, it could still build up a lot of speed. then when the glide phase starts it will not control the altitude actively anymore and could build up altitude again. 

To prevent this, the only ways to prevent this are:
- a longer approach
- a lower approach
- flaps or airbrakes on the plane

If the plane does not gain altitude and still floats too far you can modify the point of ground contact by adjusting the Land Altitude. Yes I know this sounds counter intuitive. 
The land Altitude can be negative for a very good reason. If you set a negative land altitude, that is below the ground, the path from the final approach waypoint towards the target coordinates gets lower and will intersect with the ground much sooner. This would look like this: 
![image](https://github.com/user-attachments/assets/609e93f4-b5cf-4fea-9347-6fa07908980a)

If you say that the plane has still an altitude of 15m when passing the safe home coordinates, then you can move the land altitude to negative 10-15m (start with -10 and see how it changes first) and it should make the plane touch down much sooner or perfectly on spot. 

**BUT BE CAREFUL**: If you use a Lidar, no further action is needed. The ground will trigger the glide and flare. If you have no lidar, you have to adjust the glide altitude so it starts ABOVE the ground, as that references on the Land Altitude otherwise. 
That means if your land altitude is -10m and your glide altitude is supposed to be 3m above the ground, the new glide altitude needs to be 13m. 

try these changes and then tell me if it worked for you as intended.

**Reply by rts18** [reply]
@b14ckyy Thanks for the quick reply. 

>There are things that need to be indeed optimized for planes with high glide rationes and that was already a topic during development.

Its good to know that these conditions have already been discussed for a possible future fix.
I have previously applied some of your recommendations to my aircraft that have a high glide ratio and lower wing loading. But unfortunately its too inconsistent to trust under all wind conditions. Which makes auto landing too risky for my more expensive aircraft.
So I thought it might be good to raise the matter; to see how many other dudes have the same issues.

**Reply by b14ckyy** [reply]
Different wind conditions are very hard to handle and there are a lot of limitations.
Without additional sensors like Airspeed, to actively control sink rate and speed during the final phase, a precision landing at all wind conditions is hard to achieve.

There will always be comprimises. Did you try the negative land altitude before?

**Reply by rts18** [reply]
Yes I have tried setting a negative landing altitude. It would work maybe 3 times in 5. Which isn't bad odds, but not totally reliable because of the lift variable of a given headwind. Its like tossing a coin to guess what to set it to. 10 to 20m is the general range it can be out by without a rangefinder.
If I was to set the `land altitude` to e.g. 15m. Sometimes it may still be 5m above the ground. And other times when the wind backs off; the aircraft will try to take out the fence on the approach.

**Reply by b14ckyy** [reply]
that's actually interesting and the opposite what I would expect... 

INAV has no speed control for fixed wings. every approach should have the same Airspeed and sink rate. If the wind is calm, the ground speed would be higher and the plane would fly further. With stronger headwind the ground speed would be slower so the plane would land earlier as the sink rate will be the same. 

Of course sudden gusts can give the plane a lift boost but that's something that's impossible to counteract without forcing the nose into the ground. I have tested with with my AR pro that really glides long plus has a high stall speed so its most affected by wind and with stronger winds it always lands short. But never too long. 

Do you have a few DVR recordings with OSD of some landings?

**Comment by rts18** [comment]
> that's actually interesting and the opposite what I would expect...

You're correct; most of the time its as you say. Higher winds slow the aircraft down and it falls short. But its the times the wind is gusty; that I see it the way described; when the gust lifts the aircraft from that burst. Which is the way the weather is ATM.
I had loads of DVR footage until a week back. But thought it a waste of drive space keeping failed landing attempts in HD. I only keep the good stuff I'd watch again.

**Reply by rts18** [reply]
@b14ckyy To clarify; the flight area is on a slight plateau.  Which can cause extra lift over the landing location. But that's just what I have to deal with.

**Reply by b14ckyy** [reply]
yeah this makes total sense then. Landing planes autonomously with the sensors that we have is not the easiest task for sure :D
